httpserver/https_client.cpp

```diff
@@ -50,6 +51,44 @@ void make_https_request(boost::asio::io_context& ioc,
     session->start();
 }
 
+void make_https_request(boost::asio::io_context& ioc, const std::string& url,
+                        const std::shared_ptr<request_t>& req,
+                        http_callback_t&& cb) {
+
+    static boost::asio::ip::tcp::resolver resolver(ioc);
+
+    constexpr char prefix[] = "https://";
+    std::string query = url;
+
+    if (url.find(prefix) == 0) {
+        query.erase(0, sizeof(prefix) - 1);
+    }
+
+    boost::algorithm::erase_all(query, prefix);
```

Comment:
This looks out of place? Erasing all instances of `https://` even if not at the front, seems like a in this case a really malformed url?

Reply:
Removed (It seems like I used it to remove the first instance, but then wrote the code above and forgot to delete it)